src/app/utils/bibDetailsUtils.jsx

```diff
@@ -212,6 +212,73 @@ const constructSubjectHeadingsArray = (url = '') => {
     });
 };
 
+/**
+ * isRtl(string)
+ * Returns true (or false) depending on whether the string needs to be read right to left
+ * @param {string} string
+ * @return  {boolean}
+ */
+const isRtl  = (string) => {
+  if (typeof string !== 'string') { return null }
+  return string.substring(0, 1) === '\u200F'
+}
+
+/**
+ * stringDirection(string)
+ * 'rtl' if the string needs to be read right to left, otherwise null
+ * @param {string} string
+ * @return {string}
+ */
+const stringDirection = (string) => {
+  return isRtl(string) ? 'rtl' : null
+}
+
+/**
+ * combineMatching(el1, el2)
+ * Combines properties from matching (i.e. parallel) elements as necessary
+ * Right now, this is only needed to add the 'noteType' in case of parallel notes
+ * @param {string} el1
+ * @param {object} el2
+ * @return {object}
+ */
+
+ const combineMatching = (el1, el2) => (el2 && el2.noteType) ? { noteType: el2.noteType, '@type' : el2['@type'], prefLabel: el1 } : el1;
+
+
+/**
+ * interleave(arr1, arr2)
+ * Given two arrays, returns the elements interleaved, with falsey elements removed.
+ * Example: interleave ([1, 2, null, 3], [5,6,7,8,9]) =>
+ * [1,5,2,6,7,3,8,9].
+ * Assumes that arr2 is at least as long as arr1.
+ *
+ * @param {array} arr1
+ * @param {array} arr2
+ * @return {array}
+ */
+const interleave = (arr1, arr2) => arr2.reduce((acc, el, id) => (arr1[id] && acc.push(combineMatching(arr1[id], el)), el && acc.push(el), acc), [])
+
+
+/**
+ * matchParallels(bib)
+ * Given a bib object returns a new copy of the bib in which fields with parallels have been rewritten
+ * The new rewritten field interleaves the parallel field and the paralleled (i.e. original) field together.
+ * Skips over subject fields since these require changes to SHEP
+ * @param {object} bib
+ * @return {object}
+ */
+const matchParallels = (bib) => {
+  const parallelFieldMatches = Object.keys(bib).map((key) => {
+    if (key.match(/subject/i)) { return null }
+    const match = key.match(/parallel(.)(.*)/)
+    const paralleledField = match && `${match[1].toLowerCase()}${match[2]}`
+    const paralleledValues = paralleledField && bib[paralleledField]
+    return paralleledValues && { [paralleledField] : interleave(bib[key], paralleledValues)}
+  })
+
+  return Object.assign({}, bib, ...parallelFieldMatches)
```

Comment:
_I think_ you can use `{ ...bib, ...parallelFieldMatches }` instead of `Object.assign`. Both essentially copy the objects passed into a new object, but I don't know if `bib` has complicated deeply nested objects where using just the spread operator is fine. Not a change that needs to be done but just pointing it out because `parallelFieldMatches` is being spread so `bib` could as well for the new object that is returned.

Reply:
The syntax is a little weird but in this case `parallelFieldMatches` is actually an array. The spread operator feeds all the elements of `parallelFieldMatches` to `Object.assign`. If we spread as an object `{ ...bib, ...parallelFieldMatches }` instead of getting what we want we actually get an object with a bunch of numbers as keys mixed up with the keys from the `bib` object.

Reply:
Got it, that is a bit hard to read but right, `Object.assign` is the better option here.